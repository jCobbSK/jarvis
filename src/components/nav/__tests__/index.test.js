import preact from "preact";
import render from "preact-render-to-json";

import Nav from "../index.js";

const DEFAULT_OPTIONS = {
  makers: {}
};
const MOCK_OPTIONS = {
  name: "TEST_NAME",
  version: "1.0",
  makers: {
    name: "TEST_NAME",
    email: "a@b.com",
    url: "http://"
  }
};

describe("Nav", () => {
  it("renders default attributes", () => {
    const sut = render(Nav(DEFAULT_OPTIONS));

    expect(sut).toMatchSnapshot();
  });
  it("renders all provided attributes", () => {
    const sut = render(Nav(MOCK_OPTIONS));

    expect(sut).toMatchSnapshot();
  });
});
