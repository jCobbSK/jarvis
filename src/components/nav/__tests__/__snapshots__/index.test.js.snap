// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders all provided attributes 1`] = `
<ul
  class="nav"
>
  <li
    class="project"
  >
    TEST_NAME
  </li>
  <li
    class="version"
  >
    1.0
  </li>
  <li
    class="makers"
  >
    <span>
      TEST_NAME
    </span>
    <a
      class="nav-link"
      href="mailto:a@b.com"
    >
      a@b.com
    </a>
    <a
      class="nav-link"
      href="http://"
      target="_blank"
    >
      http://
    </a>
  </li>
  <li
    class="snippets"
  >
    snippets
  </li>
</ul>
`;

exports[`Nav renders default attributes 1`] = `
<ul
  class="nav"
>
  <li
    class="project"
  />
  <li
    class="version"
  >
    NaN
  </li>
  <li
    class="makers"
  >
    <span />
  </li>
  <li
    class="snippets"
  >
    snippets
  </li>
</ul>
`;
